evalml/tests/component_tests/test_one_hot_encoder.py

```diff
@@ -428,3 +427,17 @@ def test_ohe_features_to_encode_no_col_names():
     col_names = set(X_t.columns)
     assert (col_names == expected_col_names)
     assert ([X_t[col].dtype == "uint8" for col in X_t])
+
+
+def test_ohe_top_n_categories_always_the_same():
+    df = pd.DataFrame({"categories": ["cat_1"] * 5 + ["cat_2"] * 4 + ["cat_3"] * 3 + ["cat_4"] * 3 + ["cat_5"] * 3,
+                       "numbers": range(18)})
+
+    def check_df_equality(random_state):
+        ohe = OneHotEncoder(top_n=4, random_state=random_state)
+        df1 = ohe.fit_transform(df)
+        df2 = ohe.fit_transform(df)
+        pd.testing.assert_frame_equal(df1, df2)
+
+    check_df_equality(5)
+    check_df_equality(get_random_state(5))
```

Comment:
Test looks good! Have you verified this was failing before the fix was included?

Reply:
Thanks! Yes, this test fails on `main`!